### PR TITLE
Add scripts for starting new release

### PR DESCRIPTION
**Pull request by annakocheshkova**
Things to consider before you submit the PR:

* [X] Are the files formatted correctly?
* [X] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?

## Description

The task fills the `ac-build-config.xml` file with a provided **sdkVersion** and android/apple latest release versions from GitHub.

## Related PRs or issues

[AB#74655](https://msmobilecenter.visualstudio.com/454a20a9-dfe6-4c1f-8ae8-417f76adb472/_workitems/edit/74655)

### CODE REVIEW

File: version.cake [hunk 1]
```diff
@@ -159,6 +164,47 @@ Task("StartNewVersion").Does(()=>
...
+        Information($"Nothing to replace. Exiting...");
+        return;
+    }
+    Information($"Replacing build config versions: androidVersion is {VersionReader.AndroidVersion} and iosVersion is {VersionReader.IosVersion}.");
+    ReplaceTextInFiles(ConfigFile.Path, VersionReader.AndroidVersion, androidLatestVersion);
+    ReplaceTextInFiles(ConfigFile.Path, VersionReader.IosVersion, appleLatestVersion);
+    Information($"Successfully replaced iosVersion with {appleLatestVersion} and androidVersion with {androidLatestVersion} in ac-build-config.xml.");
+}).OnError(HandleError);
+
+Task("BumpMinorVersion").Does(() =>
```
Comment (MatkovIvan) [inline version.cake]: We should increment only patch version after the release

File: version.cake [hunk 2]
```diff
@@ -159,6 +164,47 @@ Task("StartNewVersion").Does(()=>
     ReplaceRegexInFilesWithExclusion("**/Info.plist", bundleShortVersionPattern, newBundleShortVersionString, "/bin/", "/obj/", "Demo");
 });
 
+// Fills sdk, android and ios versions in the build config file with the relevant ones.
+// sdkVersion must be provided as a parameter.
```
Comment (MatkovIvan) [inline version.cake]: We already have cake target to setup `sdkVersion`, please

1. Remove `sdkVersion` from there
2. Split to Android and iOS tasks
3. Add common empty task that depends on Android and iOS ones

File: version.cake [hunk 3]
```diff
@@ -159,6 +164,57 @@ Task("StartNewVersion").Does(()=>
...
```
Comment (MatkovIvan) [inline version.cake]: Update comment

File: version.cake [hunk 4]
```diff
@@ -159,6 +164,57 @@ Task("StartNewVersion").Does(()=>
...
+Task("UpdateToLatestVersions")
```
Comment (MatkovIvan) [inline version.cake]: Update what? Consider to rename it

File: version.cake [hunk 5]
```diff
@@ -159,6 +164,57 @@ Task("StartNewVersion").Does(()=>
...
+    ParseSemVer(androidLatestVersion);
+    var versionsAreEqual = VersionReader.AndroidVersion.Equals(androidLatestVersion);
+    if (versionsAreEqual) 
+    {
+        Information($"Nothing to replace. Exiting...");
+        return;
+    }
+    Information($"Replacing build config version: androidVersion is {VersionReader.AndroidVersion}.");
+    ReplaceTextInFiles(ConfigFile.Path, VersionReader.AndroidVersion, androidLatestVersion);
+    Information($"Successfully replaced androidVersion with {androidLatestVersion} in ac-build-config.xml.");
```
Comment (MatkovIvan) [inline version.cake]: Do we have const for this file name?

File: version.cake [hunk 6]
```diff
@@ -159,6 +164,57 @@ Task("StartNewVersion").Does(()=>
...
+    var androidLatestVersion = GetLatestGitHubReleaseVersion(AndroidSdkRepoName);
+    Information($"Received latest android sdk release version {androidLatestVersion}. Verifying if it's a valid semver version...");
+    ParseSemVer(androidLatestVersion);
+    var versionsAreEqual = VersionReader.AndroidVersion.Equals(androidLatestVersion);
+    if (versionsAreEqual) 
+    {
+        Information($"Nothing to replace. Exiting...");
+        return;
+    }
+    Information($"Replacing build config version: androidVersion is {VersionReader.AndroidVersion}.");
```
Comment (MatkovIvan) [inline version.cake]: was?

Comment (annakocheshkova) [reply version.cake]: No, at this point is

File: version.cake [hunk 7]
```diff
@@ -13,6 +15,9 @@ using System.Text.RegularExpressions;
 using System.Xml.Linq;
 using System.Xml.XPath;
 
+var AndroidSdkRepoName = "microsoft/appCenter-sdk-android";
+var AppleSdkRepoName = "microsoft/appCenter-sdk-apple";
```
Comment (russelarms) [inline version.cake]: Isn't it "appcenter"?

**Comment by annakocheshkova** [comment]
@MatkovIvan No, it doesn't hang

File: scripts/configuration/config-parser.cake [hunk 8]
```diff
@@ -6,6 +6,8 @@ using System.Collections.Generic;
 public static class ConfigFile
 {
     public const string Path = "scripts/configuration/ac-build-config.xml";
```
Comment (MatkovIvan) [inline scripts/configuration/config-parser.cake]: Reuse `Name ` here

Comment (annakocheshkova) [reply scripts/configuration/config-parser.cake]: It should be a const. If I use a variable it isn't and fails build.

File: scripts/git-pull-request.sh [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+
+resp="$(curl -s -X POST $REQUEST_URL_PULL -d '{
+      "title": "Start new '${NEW_SDK_VERSION}' version",
+      "body": "Start new '${NEW_SDK_VERSION}' version",
+      "head": "release/'${NEW_SDK_VERSION}'",
+      "base": "master"
+    }')"
+url="$(echo $resp | jq -r '.url')"
+
+if [ -z $url ] || [ "$url" == "" ] || [ "$url" == "null" ]; then
```
[the remark this answers is not in the document]
Comment (annakocheshkova) [reply scripts/git-pull-request.sh]: What do you mean by positive condition? When this request failed, it returned "null", but we can't guarantee that it won't be empty.

File: version.cake [hunk 10]
```diff
@@ -159,6 +164,70 @@ Task("StartNewVersion").Does(()=>
     ReplaceRegexInFilesWithExclusion("**/Info.plist", bundleShortVersionPattern, newBundleShortVersionString, "/bin/", "/obj/", "Demo");
 });
 
+// Fills android and ios versions in the build config file with the relevant ones.
+Task("UpdateNativeVersionsToLatest")
+.IsDependentOn("UpdateSdkVersion")
```
Comment (MatkovIvan) [inline version.cake]: Should be removed from `UpdateNativeVersionsToLatest `

File: version.cake [hunk 11]
```diff
@@ -159,6 +164,70 @@ Task("StartNewVersion").Does(()=>
...
+.IsDependentOn("UpdateAndroidVersionToLatest")
+.IsDependentOn("UpdateIosVersionToLatest");
+
+Task("UpdateSdkVersion").Does(() =>
```
Comment (MatkovIvan) [inline version.cake]: No need for this task

File: version.cake [hunk 12]
```diff
@@ -159,6 +164,70 @@ Task("StartNewVersion").Does(()=>
...
+
+Task("IncreasePatchVersion").Does(() => 
+{
+    var sdkVersion = ParseSemVer(VersionReader.SdkVersion);
+    var patchVersion = sdkVersion.Patch;
+    patchVersion++;
+    sdkVersion = sdkVersion.Change(sdkVersion.Major, sdkVersion.Minor, patchVersion);
+    Information($"Bumping {VersionReader.SdkVersion} sdk version to {sdkVersion.ToString()}...");
+    UpdateConfigFileSdkVersion(sdkVersion.ToString());
+    Information($"Successfully wrote the changes to ac-build-config.xml.");
```
Comment (MatkovIvan) [inline version.cake]: Use const for name

File: version.cake [hunk 13]
```diff
@@ -159,6 +164,70 @@ Task("StartNewVersion").Does(()=>
...
+    ParseSemVer(appleLatestVersion);
+    var versionsAreEqual = VersionReader.IosVersion.Equals(appleLatestVersion);
+    if (versionsAreEqual) 
+    {
+        Information($"Nothing to replace. Exiting...");
+        return;
+    }
+    Information($"Replacing build config versions: iosVersion is {VersionReader.IosVersion}.");
+    ReplaceTextInFiles(ConfigFile.Path, VersionReader.IosVersion, appleLatestVersion);
+    Information($"Successfully replaced iosVersion with {appleLatestVersion} in ac-build-config.xml.");
```
Comment (MatkovIvan) [inline version.cake]: Use const for name

**Comment by MatkovIvan** [comment]
Squashed 50 commits to 4 and addressed feedback

**Comment by MatkovIvan** [comment]
Merging it as is to feature branch, future comments (if any) will be addressed separately

**Pull request merged**